### ISSUE TITLE
Tricky case not detected properly by MA1000

### ISSUE THREAD

**Issue by jairbubbles** [meziantou/Meziantou.Analyzer#219]
**Context**
Rule identifier: _MA0100_
Target Framework: _.NET Framework 4.7.2_
Language version: _8.0_

**Describe the bug**

It's a bit tricky but the analyzer does not seem to detect when a task is awaited in another function and not directly in the scope.

```csharp
public static async Task AnalyzerExample()
{
    // Will report an error MA0100
    using (null)
    {
        await ExecuteAsync(() => new HttpClient().GetAsync(new Uri("https://www.meziantou.net/"))).ConfigureAwait(false);
    }

    // Will not report an error
    using (null)
    {
        await ExecuteAsync(async () => await new HttpClient().GetAsync(new Uri("https://www.meziantou.net/"))).ConfigureAwait(false);
    }

    // While in both cases
    async Task ExecuteAsync(Func<Task> operation)
    {
        // we await the operation there
        await operation().ConfigureAwait(false);
    }
}
```

**Comment by meziantou** [comment]
Are you sure you are using the latest version of the analyzer? It should not report non-awaited task inside a lambda/local function.

**Comment by jairbubbles** [comment]
Probably not, I'll check!

**Comment by jairbubbles** [comment]
Nope the error is still there with version `1.0.646`.

**Comment by meziantou** [comment]
I've added a unit test in #221 with the provided code, but it doesn't reproduce your issue.

**Comment by jairbubbles** [comment]
You're right I can't reproduce in a new project.

It might be related to the fact that I get a warning when updating in the main solution:

![image](https://user-images.githubusercontent.com/15875066/107143654-84143780-6936-11eb-9b7d-556950e78cf0.png)

**Comment by jairbubbles** [comment]
And if I restart Visual Studio, the warning goes away and I no longer have the warning nor the false positive. Thx for the feedback and sorry for no trying with the latest version at the very beginning.
